app/web/src/workers/shared_webworker.ts

```diff
@@ -80,11 +83,25 @@ const dbInterface: SharedDBInterface = {
   unregisterRemote(id: string) {
     debug("unregister remote in shared", id);
     delete remotes[id];
+
+    if (Object.keys(remotes).length === 0) {
+      // Just in case there is a race between closing a tab and a new one
+      // loading, and this worker gets a new remote, don't shut down right away.
+      // Double check after a few seconds.
+      setTimeout(async () => {
+        if (Object.keys(remotes).length === 0) {
+          shutDownWebWorker();
+        }
+      }, SHUTDOWN_DELAY_MS);
+    }
   },
-  registerRemote(id: string, remote: Comlink.Remote<TabDBInterface>) {
+  async registerRemote(id: string, remote: Comlink.Remote<TabDBInterface>) {
     if (!remotes[id]) {
       debug("register remote in shared", id);
       remotes[id] = remote;
+      if (!currentRemote && (await remote.hasDbLock())) {
```

Comment:
how come we are asking if it has the lock here?

Reply:
When we reboot the shared webworker, we need to (1) re-register all the current tabs, and (2) set the current remote again. So here we check to see if the remote that was just registered has the lock. If it does, make it the current remote.